src/codegen/debug.rs

```diff
@@ -336,7 +336,7 @@ impl<'ink> DebugBuilder<'ink> {
         //Find the dimenstions as ranges
         let subscript = dimensions
             .iter()
-            .map(|it| it.get_range(index))
+            .map(|it| it.get_range_plus_one(index))
```

Comment:
we might need to do this for strings also:
```
%a = alloca [1025 x i8], align 1, !dbg !9

...

!10 = !DILocalVariable(name: "a", scope: !4, file: !5, line: 30, type: !11, align: 64)
!11 = !DICompositeType(tag: DW_TAG_array_type, baseType: !12, size: 1024, align: 64, elements: !13)
!12 = !DIBasicType(name: "char", size: 8, encoding: DW_ATE_UTF, flags: DIFlagPublic)
!13 = !{!14}
!14 = !DISubrange(count: 1024, lowerBound: 0)
``

Reply:
Good catch, done. Just to make sure though is this necessary for strings? cc @ghaith

Reply:
Not certain.. the strings have the size increased by 1 to account for the null terminator. But i'm not sure if this is something to also reflect in the debug info. I'd say we keep it and check how it behaves on debugging.
We had an internal issue at some point where strings were shown as arrays so we might eventually switch how strings are displayed.